stream-chat-android-ui-common/src/test/kotlin/com/getstream/sdk/chat/viewmodel/messages/MessageListViewModelTest.kt

```diff
@@ -30,196 +60,216 @@ private val THREAD_MESSAGES = createMessageList {
+        whenever(it.channelMutes) doReturn MutableStateFlow(listOf())
+        whenever(it.typingUpdates) doReturn MutableStateFlow(TypingEvent(channelId = CID, listOf()))
+    }
+
+    private val chatClient: ChatClient = MockChatClientBuilder {
+        mock {
+            whenever(it.queryChannel(any(), any(), any())) doReturn Channel().asCall()
+            whenever(it.notifications) doReturn mock()
+            whenever(it.flagMessage(any())) doReturn mock()
+            whenever(it.deleteMessage(any(), any())) doReturn mock()
+        }
+    }.build()
+
+    init {
+        StateRegistry.instance = stateRegistry
+    }
+
+    @Test
+    @Disabled("Can not be tested until we use mockk or other way to mock static function")
+    fun `Should request more messages when end region reached`() {
+        // given
+        val viewModel = MessageListViewModel(CID, chatClient = chatClient)
+        viewModel.state.observeAll()
+
+        // when
+        viewModel.onEvent(MessageListViewModel.Event.EndRegionReached)
+
+        // should
+        verify(chatClient).loadOlderMessages(CID, DEFAULT_MESSAGES_LIMIT)
+    }
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    @Test
+    @Disabled("asLiveData() does not perform reliably during tests")
+    fun `Should display progressbar and messages`() = runBlockingTest {
+        // given
+        val viewModel = MessageListViewModel(CID, chatClient = chatClient, globalState = globalState)
+        val stateList = viewModel.state.observeAll()
+
+        // should
+        stateList.first() shouldBeEqualTo MessageListViewModel.State.Loading
+
+        // when
+        messagesState.emit(MessagesState.Result(MESSAGES))
+        advanceUntilIdle()
+
+        // should
+        stateList.last().apply {
+            this shouldBeInstanceOf MessageListViewModel.State.Result::class
+            val state = (this as MessageListViewModel.State.Result)
+            state.messageListItem.items.filterIsInstance<MessageListItem.MessageItem>().map {
+                it.message
+            } shouldBeEqualTo MESSAGES
+        }
+    }
+
+    @Test
+    fun `When delete event doesn't have hard flag Should delete message`() {
+        // given
+        val viewModel = MessageListViewModel(CID, chatClient = chatClient)
+        viewModel.state.observeAll()
+
+        // when
+        viewModel.onEvent(MessageListViewModel.Event.DeleteMessage(MESSAGE, hard = false))
+
+        // should
+        verify(chatClient).deleteMessage(MESSAGE.id, false)
+    }
+
+    @Test
+    fun `When delete event has hard flag Should hard delete message`() {
+        // given
+        val viewModel = MessageListViewModel(CID, chatClient = chatClient)
+        viewModel.state.observeAll()
+
+        // when
+        viewModel.onEvent(MessageListViewModel.Event.DeleteMessage(MESSAGE, hard = true))
+
+        // should
+        verify(chatClient).deleteMessage(MESSAGE.id, true)
+    }
+
+    @Test
+    fun `Should flag message`() {
```

Comment:
we have a lot of inconsistent naming here, we should probably stick to one approach.